pkg/admissioncontroller/webhook/admission/resourcesize/handler.go

```diff
@@ -155,14 +218,56 @@ func isUnrestrictedUser(userInfo authenticationv1.UserInfo, subjects []rbacv1.Su
 	return userMatch(userInfo, subjects)
 }
 
-func findLimitForGVR(limits []admissioncontrollerconfigv1alpha1.ResourceLimit, gvr *metav1.GroupVersionResource) *resource.Quantity {
+func findRestrictionsForGVR(limits []admissioncontrollerconfigv1alpha1.ResourceLimit, gvr *metav1.GroupVersionResource) (*resource.Quantity, *int64) {
 	for _, limit := range limits {
 		size := limit.Size
+		count := limit.Count
 		if admissioncontrollerhelper.APIGroupMatches(limit, gvr.Group) &&
 			admissioncontrollerhelper.VersionMatches(limit, gvr.Version) &&
 			admissioncontrollerhelper.ResourceMatches(limit, gvr.Resource) {
-			return &size
+			return size, count
 		}
 	}
-	return nil
+	return nil, nil
+}
+
+func (h *Handler) getKindFromGVR(gvr *metav1.GroupVersionResource) (string, error) {
+	if h.RESTMapper == nil {
+		return "", errors.New("RESTMapper not initialized")
+	}
+
+	gvk, err := h.RESTMapper.KindFor(schema.GroupVersionResource{
+		Group:    gvr.Group,
+		Version:  gvr.Version,
+		Resource: gvr.Resource,
+	})
+	if err != nil {
+		return "", err
+	}
+	return gvk.Kind, nil
+}
+
+func (h *Handler) existingResourcesExceedLimit(_ admission.Request, gvr *metav1.GroupVersionResource, limit int64) (bool, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
```

Comment:
Why don't we pass the `context.Context` from the `Handle` function down here? This would be better instead of using `context.Background`